app/src/main/kotlin/com/wire/android/ui/home/HomeNavigation.kt

```diff
@@ -52,27 +63,27 @@ sealed class HomeNavigationItem(
     @StringRes val title: Int,
     val isSearchable: Boolean = false,
     val isSwipeable: Boolean = true,
-    val content: @Composable (NavBackStackEntry) -> Unit
+    val content: (UpdateScrollPosition) -> (HomeNavigationContent)
 ) {
 
     object Conversations : HomeNavigationItem(
         route = HomeDestinations.conversations,
         title = R.string.conversations_screen_title,
         isSearchable = true,
         isSwipeable = false,
-        content = { ConversationRouter() },
+        content = { updateScrollPosition -> { ConversationRouter(updateScrollPosition = updateScrollPosition) } }
     )
 
     object Vault : HomeNavigationItem(
         route = HomeDestinations.vault,
         title = R.string.vault_screen_title,
-        content = { VaultScreen() },
+        content = { { VaultScreen() } }
     )
 
     object Archive : HomeNavigationItem(
         route = HomeDestinations.archive,
         title = R.string.archive_screen_title,
-        content = { ArchiveScreen() },
+        content = { { ArchiveScreen() } }
```

Comment:
I am not sure if it is a good thing to pass typealias UpdateScrollPosition = (Int) -> Unit as a parameter to the lambda but do nothing with it, is this updateScrollPOsition functionality going to be used for all the views in the future, if not maybe we have to distinct it somehow with the typealias that would not take this typealias UpdateScrollPosition = (Int) -> Unit  as parameter ? also not really know a quick fix for this one

Reply:
as far as we don't have a designs for Archive and Vault we can try to guess :) 
I believe we'll have search field in Archive too, so I assume - it will be hidable there too, so it will be used there.

Reply:
>I am not sure if it is a good idea to pass the updateScrollPosition() lambda all the way to each view, that should update the scroll position, but I am not sure how we could solve it, one way I could think of is to use compositionLocalOf() which would provide some object that is accesible from the whole app and would update the scroll position of the HomeScreen, but it is just a quick guess, maybe you could try to find a way to omit passing the scrollposition callback alll the way through each view ?

☝️ I attached also this comment as it feels connected and here we can reply 😄 

The approach with `staticCompositionLocalOf` I think should be used for things that are used throughout the whole app, so I'm not sure - passing it all the way to the view ensures that we always use the correct one and that we always have it handled, but if it's really going to be more widely used across a larger part of the app than just this particular case then it might be worth trying at least. We can make it versatile - every child that can scroll should use such local composition and update its scroll value and every parent that wants to make use of such value should provide its own implementation of it which would then be available to all his children. The problem with this is that it can be provided again by the child and then the scroll updates will no longer reach the original parent. What do you think, is it worth trying?

Reply:
True you are probably right :p I dont not know if local conpisition is the best approach here just was giving a guess to maybe a solution that could work but i agree with you. I just have a feeling that there could be a better solution to assign the scroll position with less overhead but if passing the lamda like this is the only option we got I am fine with it :p just my Gut feeling saying but i trust Boris on this one :p

Reply:
Btw. I like your approach maybe we can just remember not to provide composition local second time :p

Reply:
forget it , :P boris already handled this discussion in the newest PR

Reply:
you mean in the latest commit? 😄  
I'm fine also with the approach Boris proposed, where dependency injection is used instead of local composition